### ISSUE TITLE
Ps1 issues on a chromebook

### ISSUE THREAD

**Issue by Senshi209** [BinBashBanana/webretro#40]
Whenever I upload the rom for Metal Gear Solid on my Chromebook, it crashes and I have to power refresh. Sometimes it gets to the "Select Master ROM screen" before crashing, and when I select a .cue or .m3u file it crashes. Can my Chromebook just not run games made on the Playstation, or is there a fix for this? I really want to emulate the Playstation on my Chromebook.

**Comment by ethanaobrien** [comment]
@BinBashBanana I've noticed this in the new cores in my repository too (although, I've only experienced it with the fceumm core) so there's definitely some cause, probably within retroarch itself.

**Comment by BinBashBanana** [comment]
If your chromebook itself is crashing, the issue is probably not enough RAM. Can you also test with the Mupen64Plus-Next core?

**Comment by HaxsMaster** [comment]
@Senshi209 If you wouldn't mind, can you give me some advice on finding PS1 ROMS and running them?

**Comment by ethanaobrien** [comment]
@Senshi209 just wondering, does the tab crash (does it give an out of memory error?), does the emulator just freeze? Or does something else happen

**Comment by Senshi209** [comment]
Tested the Mupen64Plus-Next Core with Ocarina of Time and Goldeneye 007. They work, however sometimes the emulator freezes and I refresh the page. Mysteriously, with the ps1 emulator, my CHROMEBOOK crashes, not just the emulator. To play n64 I usually use the other n64 core, ParaLLEI. It doesn't make the emulator freeze in between and I am currently doing a playthrough of Ocarina of Time on it, the emulator works. Sometimes there are minor texture glitches, but ParaLLEI works.


HaxsMaster, to upload a rom, you should have a folder with the bin and cue files for the game you would like to play. For multi-disk games, simply add the bin and cue files for each disk and put them in the same folder. You also have to create an m3u file, which you can find how to do here(this is for another emulator called OpenEmu however it teaches you how to create m3u files)https://gist.github.com/emilyst/4cbe6a11739bd2c43e9418322a9575ce
When you have your folder, then select choose rom file in webretro and when it says upload directory, upload the folder you created for the game you wanted to play. Sorry if its kind of confusing.

**Comment by Senshi209** [comment]
Do you guys know why the Mupen64Plus-Next Core works but not Beetle PSX?

**Comment by HaxsMaster** [comment]
@Senshi209 I suspect it may have something to do with the sheer limitations of a chromebook. But where can I find consistent downloads for those .bin and .cue files? If they were ZIP archives, that would be greatly convienient.

**Comment by Senshi209** [comment]
@HaxsMaster The sheer limitations of a chromebook would make sense, but then how come it runs Mupen64Plus-Next? It doesn't make sense, because if this was because of specs, then n64 wouldn't run on my chromebook but it does. @BinBashBanana, is this a problem with retroarch itself? Because it doesn't make sense that a chromebook runs n64 but not ps1. That too, it runs the Mupen64Plus-Next n64 core.

@HaxsMaster here is where I usually download ps1 roms: https://vimm.net/vault/PS1

It has a ton of roms, and I use it to download games for my emulators and they put their ps1 roms in .7z files. However, for a multi disk game you will have to download the .7z files seperately. There is a drop down menu next to the download button which allows you to choose which disk you would like to download.

**Comment by HaxsMaster** [comment]
@Senshi209 You can run games like Mario 64 on chromebook, though it may be a specialized version since it was apparently made in HTML. And you can run N64 games at 60 frames. However, the limitations of the system itself result in the game running at 30 frames, which is the max.

**Comment by Senshi209** [comment]
Yes, I beat the HTML version of sm64 on my chromebook. However, I am playing OoT for the n64 through the emulator on my chromebook, and it works very well. So why doesn't ps1 work?

**Comment by HaxsMaster** [comment]
Perhaps due to the amount of data that the chromebook has to process? From my knowledge, PS1 games were huge compared to N64 games, especially since they used CDs rather than cartridges. Are the download sizes any different? And also, any help on unzipping the .7z files since I'm also on Chromebook?

**Comment by Senshi209** [comment]
I'm not sure if its because of the size of the file, because I think that just depends on how much space you have left on your computer. Plus, the n64 was more powerful than the ps1, so if n64 games run and not ps1 then im guessing its some kind of bug. 

Try this for the .7z files: https://extract.me/

**Comment by HaxsMaster** [comment]
Interesting note that the N64 was more powerful. I'll have to fact check that, but yeah, I'd agree that it is probably more dependent on the space left.

So I just have to create a separate folder for the .bin and .cue files once I have uncompressed them, and use that as a directory for the emulator, right?

**Comment by Senshi209** [comment]
No, not seperate folders. Put all the files in the same folder.

**Comment by Senshi209** [comment]
Has anybody figured out what's happening? If it helps, I am using a Dell Chromebook 3100.

**Comment by ethanaobrien** [comment]
Chances are your chromebook literally doesn't have enough memory, there's no way to really fix this, sorry. The reason why n64 works is because the roms are much smaller, so PlayStation takes more memory. Sorry about this

**Comment by HaxsMaster** [comment]
@Senshi209 When I used the extraction tool (on chromebook) It ended up freezing at 7 to 8%. How long does it usually take to unzip .7z files?

**Comment by sDetective** [comment]
Yeah I've gotten these issues for PlayStation core also, I get weird graphical issues where the colors are on the fritz and the fps are low or just plainly doesn't work.

**Comment by BinBashBanana** [comment]
Can you go to ChromeOS diagnostics and tell me how much RAM (memory) you have?

**Comment by sDetective** [comment]
Its a school chromebook I don't think I could oh nvm I have it

**Comment by sDetective** [comment]
![image](https://user-images.githubusercontent.com/85586892/193289646-69e8c77b-3cf6-4ca8-b54b-7a407f815c86.png) Here it is

**Comment by sDetective** [comment]
how do I free this up?

**Comment by sDetective** [comment]
I have a lot of tabs open rn thats why its near max

**Comment by sDetective** [comment]
![image](https://user-images.githubusercontent.com/85586892/193293546-10066988-92f7-4c9e-ad71-3951985738a8.png)
This is re3, I've tried messing with the settings to solve this issue but nothing seems to work, It happened with metal gear solid too so I'm assuming other PlayStation games are bugged as well.

**Comment by BinBashBanana** [comment]
@sDetective specs look ok, and the funny graphics can be fixed by going to "More" > "Pre-start options" > "Use software renderer" in the menu bar before choosing the ROM.

**Comment by sDetective** [comment]
its not there 
![image](https://user-images.githubusercontent.com/85586892/193360808-9637d21a-2784-4979-b2eb-cac2f9ba3d5a.png)
These are the only thing available

**Comment by sDetective** [comment]
Oh wait do you mean in the main retroarch settings?

**Comment by sDetective** [comment]
yep that fixed the issue, I didn't know about the menu bar appeared before rom selection

**Comment by RottenEggNog** [comment]
> Whenever I upload the rom for Metal Gear Solid on my Chromebook, it crashes and I have to power refresh. Sometimes it gets to the "Select Master ROM screen" before crashing, and when I select a .cue or .m3u file it crashes. Can my Chromebook just not run games made on the Playstation, or is there a fix for this? I really want to emulate the Playstation on my Chromebook.

How are you able to use this on chromebook?
